README.md

```diff
@@ -1,2 +1,51 @@
-# spark
-Spark is an implement of Axon kicker.
+# Spark
+Spark is responsible for the following tasks:
+- Constructing CKB transactions to implement Axon’s staking mechanism. The related code can be found in the `tx-builder` directory.  
+- Providing query APIs for Axon’s frontend to display information to end users. For example, details such as stake, delegate, and reward history of users. The related code can be found in the `rpc-client` directory.  
+- Recovering Sparse Merkle Trees (SMTs).  
+
+The corresponding architecture is depicted below.
+![spark_arch](./spark_arch.png)
+
+To learn more about Axon’s staking and kicker, please visit [staking protocol](https://docs.axonweb3.io/staking/staking-protocol).
+We will describe the three tasks mentioned above in detail.
+
+# CKB Transaction Construction
+Transaction construction can be divided into three categories:
+
+## Staker
+These transactions are initiated by stakers who are responsible for paying the gas fees. Examples include increasing or decreasing a staker’s stake.  
+
+## Delegator
+These transactions are initiated by delegators who are responsible for paying the gas fees. Examples include increasing or decreasing a delegator’s delegation to a staker.  
+## Kicker
+Within the Axon ecosystem, a kicker plays the permissionless role to address various critical tasks, including:
+
+- Monitoring the liveness of any Axon-based chain and promptly reporting its status (aka., checkpoint) to Layer 1.
+- Facilitating the selection for the upcoming set of validators (aka., metadata) when the previous tenure concludes.
+- Ensuring the accuracy and timeliness of the information regarding the latest staking and delegating allocations.
+
+This section will construct transactions to implement these functions.
+
+## Workflow
+For transactions initiated by a staker or delegator, the process must be started by Axon’s frontend (aka `front` in the following picture), built by spark, and finally executed in CKB.  
+Here is the entire workflow:  
+![front_spark_ckb](./front_spark_ckb.png)  
+For transactions initiated by Kicker (submitting checkpoint, updating metadata to CKB, ...), there is no interaction between the frontend and spark.
+
+# Query
+This can be divided into two categories:
+
+## Individual Staker or Delegator
+- The stake, delegate, reward transaction history of a staker or delegator.
+- The total stake or delegate amount.
+- ...
+## Axon Chain
+- Total stake, delegate amount.
+- Epoch, period, block number.
+- Top stake address
+- ...
+
+# SMT Recovery
+Anyone who wants to run a kicker will need to query CKB and an existing kicker to recover the necessary SMTs.
+SMTs are essential for constructing many types of transactions.
```

Comment:
To run a kicker, one must query CKB and an existing kicker to recover the necessary SMTs, which are essential for constructing various transaction types.